ambry-store/src/test/java/com.github.ambry.store/CompactionPolicyTest.java

```diff
@@ -230,17 +229,22 @@ public void testDifferentMessageRetentionDays() throws StoreException, Interrupt
   }
 
   /**
-   * Generates random strings
-   * @param count the total number of random strings that needs to be generated
-   * @return a {@link List} of random strings of size {@code count}
+   * Generates random log segment names
+   * @param count the total number of random log segment names that needs to be generated
+   * @return a {@link List} of random log segment name of size {@code count}
    */
-  static List<String> generateRandomStrings(int count) {
-    List<String> randomStrings = new ArrayList<>();
-    for (int i = 0; i < count; i++) {
-      randomStrings.add(UtilsTest.getRandomString(5));
+  static List<String> generateRandomLogSegmentName(int count) {
+    List<String> randomLogSegmentNames = new ArrayList<>();
+    while (randomLogSegmentNames.size() < count) {
+      String logSegmentNames =
```

Comment:
1. Generate two longs (or one long) and use `LogSegmentNameHelper.getName()`
2. Make sure that the longs being generated don't repeat (this you already do by checking the names)
3. Set the limit to (which is now 20) to something like `1000 * count`